src/actions/posts.ts

```diff
@@ -429,9 +428,11 @@ export function setUnreadPost(userId: string, postId: string) {
                 channelId: post.channel_id,
             },
         });
-
         try {
-            unreadChan = await Client4.markPostAsUnread(userId, postId);
+            if (post.system_post_ids) {
```

Comment:
We should probably do this before we dispatch the ADD_MANUALLY_UNREAD. Dispatching that adds some behaviour that stops the channel from being marked unread, so if we're preventing you from marking the system message unread, we also should avoid doing that.